language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java

```diff
@@ -260,16 +276,53 @@ private Optional<String> getVariableName(Node matchedNode) {
                 ConstantDeclarationNode constantDecl = (ConstantDeclarationNode) matchedNode;
                 return Optional.of(constantDecl.variableName().text());
             case OBJECT_FIELD:
-                ObjectFieldNode objectFieldNode = (ObjectFieldNode) matchedNode;
-                return Optional.of(objectFieldNode.fieldName().text());
+                return getObjectFieldName((ObjectFieldNode) matchedNode);
             case LET_VAR_DECL:
-                LetVariableDeclarationNode variableDecl = (LetVariableDeclarationNode) matchedNode;
-                BindingPatternNode node = variableDecl.typedBindingPattern().bindingPattern();
-                return Optional.of(((CaptureBindingPatternNode) node).variableName().text());
+                return getLetVarName((LetVariableDeclarationNode) matchedNode);
+            case LET_EXPRESSION:
+                Node parent = matchedNode.parent();
+                switch (parent.kind()) {
+                    case LOCAL_VAR_DECL:
+                        return getLocalVarName((VariableDeclarationNode) parent);
+                    case MODULE_VAR_DECL:
+                        return getModuleVarName((ModuleVariableDeclarationNode) parent);
+                    case OBJECT_FIELD:
+                        return getObjectFieldName((ObjectFieldNode) parent);    
+                    case LET_VAR_DECL:
+                        return getLetVarName((LetVariableDeclarationNode) parent);
+                    default:
+                        return Optional.empty();
```

Comment:
We can replace this by ` return getVariableName(parent)`

Reply:
Here we don't want to consider `constant-decl` case. Therefore we cannot replace this.